lib/chess/functional/cell_notation.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../core/chess'
+
+module Chess
+  module Functional
+    module CellNotation
+      Algebraic = Struct.new(:column, :row) do
+        def to_s
+          "#{column}#{row}"
+        end
+
+        def to_cartesian_a
+          [row - 1, COLUMNS.index(column)]
+        end
+      end
+
+      Cartesian = Struct.new(:column, :row) do
+        def to_a
+          [row, column]
+        end
+      end
+
+      def algebraic_to_cartesian_a(name)
+        splitted_name = name.split('')
+
+        Algebraic.new(splitted_name[0], splitted_name[1].to_i).to_cartesian_a
+      end
```

Comment:
I suggest calling this method just `cartesian_coordinates`

Reply:
but you don't return an `Algebraic` object. Instead, you return an array of cartesian coordinates (the result of `.to_cartesian_a`)

Reply:
no no, here we'll need an algebraic

Reply:
hahahaha